controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -266,7 +266,8 @@ func (w *Workload) UpdateEtcdVersionInKubeadmConfigMap(ctx context.Context, imag
 		return err
 	}
 	config := &kubeadmConfig{ConfigMap: kubeadmConfigMap}
-	if err := config.UpdateEtcdMeta(imageRepository, imageTag); err != nil {
+	changed, err := config.UpdateEtcdMeta(imageRepository, imageTag)
+	if !changed || err != nil {
```

Comment:
While we can read `config.UpdateEtcdMeta` to know that it will never return both changed == true and err != nil, this is more clear to someone who hasn't read the implementation that we aren't potentially ignoring an error.
```suggestion
	if err != nil || !changed {
```